### DISCUSSION TITLE
Change the name of the photo file what we want in QFIELD

### DISCUSSION THREAD

**Discussion by Bagaszed** [opengisch/QField#5257] [Q&A]
### Describe the issue
[How to name a photo according to the field we want at QFIELD?  For example, I have a photo code field that contains the code "J0001" onwards, because every resulting image file name in qfield seems to be automatically named using the date it was taken]

### Expected behavior
[I want the name of the photo after export according to the field without changing one by one, how do I do it?  I think this uses a coding system. I'm not very good at it]

### Desktop (please complete the following information)
 - OS: [e.g. Windows]
 - QGIS Version [e.g. 3.22.0]
 - QFieldSync Version [e.g.4.0.0]

### Mobile (please complete the following information)
 - Device: [e.g. Xiaomi Mi 8]
 - OS: [e.g. Android 10]
 - QField version: [3.2.2]

### Additional information
* Problem started happening recently, didn't happen in an older version of QField: [Yes]
* Problem can be reliably reproduced, doesn't happen randomly: [Yes]
* Problem happens with all files and projects, not only some files or projects: [No]

**Comment by Ipickedausername** [comment]
Hi, 
 
When you have a field that allows taking pictures using the 'Attachment' widget, you can configure the attachments in the properties of the layer (QFIELD tab).

From there, you can use an expression to name your pictures. For example, you can use the field that contain your "codes". 

![Picture1](https://github.com/opengisch/QField/assets/72577399/1b9afb05-f4de-4a4d-a8d5-eb6facc63fa0)

**Comment by Bagaszed** [comment]
![Screenshot (749)](https://github.com/opengisch/QField/assets/70121303/2936d918-b895-4958-9d6d-d9b1efac293e)

for example i want to do that can QFIELD do that?, i mean the name of file image automaticly fill like that

**Comment by Bagaszed** [comment]
> Hi,
> 
> When you have a field that allows taking pictures using the 'Attachment' widget, you can configure the attachments in the properties of the layer (QFIELD tab).
> 
> From there, you can use an expression to name your pictures. For example, you can use the field that contain your "codes".
> 
> ![Picture1](https://private-user-images.githubusercontent.com/72577399/331889229-1b9afb05-f4de-4a4d-a8d5-eb6facc63fa0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYxNDM3NDAsIm5iZiI6MTcxNjE0MzQ0MCwicGF0aCI6Ii83MjU3NzM5OS8zMzE4ODkyMjktMWI5YWZiMDUtZjRkZS00YTRkLWE4ZDUtZWI2ZmFjYzYzZmEwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTE5VDE4MzA0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdlNzQ1OGQ5MWNjZGI3YTEwNmY4N2VkMmFlMWM2NGQ4ZGQ5NTA5YWFhNjBjNTU4YmFjYWNiNDQzYzJlNmEwMDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.VY5_zD4tbBleLWLXuVe0Ydq-rG7eCFu-XSFax_-ZEIQ)

I've done that, but the result is the same :(
![Screenshot_2024-05-20-01-30-06-244_ch opengis qfield](https://github.com/opengisch/QField/assets/70121303/e830d9e0-def1-4ce7-915b-9a0f4bda0e76)

**Comment by Ipickedausername** [comment]
Hello, 

I tried with a test project and I have the same issue indeed. 
`'DCIM/suivi-pt_' || Concat("name", "FOTO_TGH", "FOTO_UIG", "KD_FOTO") || '.{extension}'`

I add a test project for a Dev who certainly can do a better check. 

[test_tabs_Qfield (2).zip](https://github.com/opengisch/QField/files/15381752/test_tabs_Qfield.2.zip)

**Comment by roydenyates** [comment]
I am trouble shooting a similar naming issue. Pics are default named and not by the expression. I noticed however, that if I created the feature containing the pic record (i.e. saved it), then selected it and edited it to  add a pic, the renaming worked as coded. Anyone else?

I see [here](https://github.com/opengisch/QField/discussions/5222#discussioncomment-9330047) that someone else mentions saving first...

**Comment by roydenyates** [comment]
On the phone/tablet go to Qfield settings and enable "Fast Editing Mode".

That solved the renaming issue for me, as the expression with variables is then honoured.

**Comment by nirvn** [comment]
I've tested photo naming using the sample bees project with both native and non native camera on Android and couldn't replicate this issue. It'll likely take someone to share a small project that shows the issue to investigate this further.
